tests/test_slots.py

```diff
@@ -0,0 +1,351 @@
+"""
+Tests focusing on the Python part of slots.
+For tests focusing on the `{% slot %}` tag, see `test_templatetags_slot_fill.py`
+"""
+
+import re
+from typing import Dict
+
+import pytest
+from django.template import Context, Template, TemplateSyntaxError
+from django.template.base import NodeList, TextNode
+from pytest_django.asserts import assertHTMLEqual
+
+from django_components import Component, register, types
+from django_components.slots import Slot, SlotRef
+
+from django_components.testing import djc_test
+from .testutils import PARAMETRIZE_CONTEXT_BEHAVIOR, setup_test_config
+
+setup_test_config({"autodiscover": False})
+
+
+# Test interaction of the `Slot` instances with Component rendering
+@djc_test
+class TestSlot:
+    @djc_test(
+        parametrize=(
+            ["components_settings", "is_isolated"],
+            [
+                [{"context_behavior": "django"}, False],
+                [{"context_behavior": "isolated"}, True],
+            ],
+            ["django", "isolated"],
+        )
+    )
+    def test_render_slot_as_func(self, components_settings, is_isolated):
```

Comment:
These new tests check that the `Slot` instance has its fields correctly populated when instantiated in different ways